components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -387,9 +385,14 @@ func (o *odometry) trackPosition(ctx context.Context) {
 			left := positions[0]
 			right := positions[1]
 
+			props, err := o.base.Properties(ctx, nil)
+			if err != nil {
+				o.logger.Error(err)
+				return
+			}
 			// Difference in the left and right motors since the last iteration, in mm.
-			leftDist := (left - o.lastLeftPos) * o.wheelCircumference
-			rightDist := (right - o.lastRightPos) * o.wheelCircumference
+			leftDist := (left - o.lastLeftPos) * props.WheelCircumferenceMeters
+			rightDist := (right - o.lastRightPos) * props.WheelCircumferenceMeters
```

Comment:
Let's add a helper function that does the following:
- checks the current o.base's properties against a newBase's properties. If they have not changed, do nothing.
- warns the user that we're updating the properties used in a calculation, and that they might want to reset the odometry using a DoCommand.
- stores the new base in o.base and outputs the properties needed.

Reply:
there's never really a "new" base since we are always just checking the properties of o.base. I think in order to check if there's been a change we would again have to store width and circumference on the struct, is that what you mean?

Reply:
that's fine - the ask is to expose to the user when a change happens in the wheeled base dimensions, so they're not confused about a sudden difference in output from the calculation algorithm, and they can verify that their changes have in fact been processed by our code when they input new dimensions, without having to back-calculate their new outputs.